explainability-service/src/main/java/org/kie/trustyai/service/data/cache/MetricCalculationCacheKeyGen.java

```
@@ -0,0 +1,24 @@
+package org.kie.trustyai.service.data.cache;
+
+import java.lang.reflect.Method;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.kie.trustyai.service.data.storage.Storage;
+
+import io.quarkus.cache.CacheKeyGenerator;
+import io.quarkus.cache.CompositeCacheKey;
+
+@ApplicationScoped
+public class MetricCalculationCacheKeyGen implements CacheKeyGenerator {
+
+    @Inject
+    Instance<Storage> storage;
+
+    @Override
+    public Object generate(Method method, Object... methodParams) {
+        return new CompositeCacheKey(storage.get().getLastModified(), methodParams[1].hashCode());
```

`methodParams` could be empty. We can decide to fallback to random value to force cache reload in this case or throw an error if it should never happen.
It also makes the assumption (as far as I can see) that the first param is a `DataFrame`.
We might implement a test that uses Jandex to get all method annotated with `CacheResult ` that are using `MetricCalculationCacheKeyGen` to verify that the first parameter exists and is a `DataFrame`.
Wdyt?

Note: feel free to postpone to a future ticket this improvement

---

@danielezonca thanks. discussed offline.